File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        result = []
        for i in range(len(nums1) ):
            for j in range(len(nums2 )):
                if(nums1[i] == nums2[j] and nums1[i] not in result):
                    result.append(nums1[i])
                    break
        return result       

File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        Set<Integer> set1 = new HashSet<>();
        for(int n : nums1){
            set1.add(n);
        }
        Set<Integer> intersect = new HashSet<>();
        for(int n : nums2){
            if(set1.contains(n)){
                intersect.add(n);
            }
        }
        int[] arr = new int[intersect.size()];
        int i = 0;
        for(int n : intersect){
            arr[i++] = n;
        }
        return arr;
    }
}